Update README/Contributing to install openssl

Seeing this with a fresh install:

```bash
matthew@Matthews-MacBook-Pro:~/Documents/rusoto$ cargo build
   Compiling openssl-sys-extras v0.7.1
   Compiling num v0.1.28
   Compiling rust-crypto v0.2.34
   Compiling uuid v0.1.18
Build failed, waiting for other jobs to finish...
failed to run custom build command for `openssl-sys-extras v0.7.1`
Process didn't exit successfully: `/Users/matthew/Documents/rusoto/target/debug/build/openssl-sys-extras-e38fd8aa704c6185/build-script-build` (exit code: 101)
--- stdout
TARGET = Some("x86_64-apple-darwin")
OPT_LEVEL = Some("0")
PROFILE = Some("debug")
TARGET = Some("x86_64-apple-darwin")
debug=true opt-level=0
HOST = Some("x86_64-apple-darwin")
TARGET = Some("x86_64-apple-darwin")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CC_x86_64-apple-darwin = None
CC_x86_64_apple_darwin = None
HOST_CC = None
CC = None
HOST = Some("x86_64-apple-darwin")
TARGET = Some("x86_64-apple-darwin")
HOST = Some("x86_64-apple-darwin")
CFLAGS_x86_64-apple-darwin = None
CFLAGS_x86_64_apple_darwin = None
HOST_CFLAGS = None
CFLAGS = None
running: "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-g" "-m64" "-fPIC" "-o" "/Users/matthew/Documents/rusoto/target/debug/build/openssl-sys-extras-e38fd8aa704c6185/out/src/openssl_shim.o" "-c" "src/openssl_shim.c"
ExitStatus(Code(1))


command did not execute successfully, got: exit code: 1



--- stderr
src/openssl_shim.c:1:10: fatal error: 'openssl/hmac.h' file not found
#include <openssl/hmac.h>
         ^
1 error generated.
thread '<main>' panicked at 'explicit panic', /Users/matthew/.cargo/registry/src/github.com-0a35038f75765ae4/gcc-0.3.20/src/lib.rs:772
```

Did a `brew install openssl` then re-ran `cargo build` but the build failed again.  Ended up having to run `brew link --force openssl` then `cargo build`.

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -6,7 +6,7 @@ Install Rust **1.3.0** or later - http://www.rust-lang.org/
 
 Check out code from github.
 
-`pip install --user -r codegen/requirements.txt` to for Python codegen requirements.
+`pip install --user -r codegen/requirements.txt` to install Python codegen required libraries.
 
 If you're on OSX, you'll probably need a new version of openssl.  Run `brew install openssl`.  
 
@@ -29,8 +29,9 @@ Initialize and fetch the git submodule for botocore definitions:
 
 `git submodule update`
 
-Initial setup complete, the above shouldn't be needed again unless you need to
-update the botocore definitions from that upstream project.
+Now run `cargo build` to perform the initial build.
+
+Initial setup is now complete, the above shouldn't be needed again unless you need to update the botocore definitions from that upstream project.
 
 ### Building after initial setup
 

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -1,6 +1,6 @@
 # Contributing to Rusoto
 
-## Setting up build environment (only needed once)
+### Setting up build environment (only needed once)
 
 Install Rust **1.3.0** or later - http://www.rust-lang.org/
 
@@ -32,14 +32,15 @@ Initialize and fetch the git submodule for botocore definitions:
 Initial setup complete, the above shouldn't be needed again unless you need to
 update the botocore definitions from that upstream project.
 
-## Building after initial setup
+### Building after initial setup
 
 Build the project with `cargo build`.
 
-`cargo test --verbose --features aws_integration` - This will create real AWS resources and you may be charged.
+Integration tests can be executed by running `cargo test --verbose --features aws_integration`.
+This will create real AWS resources and you may be charged.
 
 For more verbose test output, you can run `cargo test --verbose --features aws_integration -- --nocapture`.
 
-## Rust code generation from boto core service definitions:
+### Rust code generation from boto core service definitions:
 
 See [CODEGEN](codegen/CODEGEN.md).

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -1,0 +1,45 @@
+# Contributing to Rusoto
+
+## Setting up build environment (only needed once)
+
+Install Rust **1.3.0** or later - http://www.rust-lang.org/
+
+Check out code from github.
+
+`pip install --user -r codegen/requirements.txt` to for Python codegen requirements.
+
+If you're on OSX, you'll probably need a new version of openssl.  Run `brew install openssl`.  
+
+If using pre-El Capitan OSX, run `brew link --force openssl`.  
+
+For El Capitan, run:
+
+```bash
+export OPENSSL_INCLUDE_DIR=`brew --prefix openssl`/include
+export OPENSSL_LIB_DIR=`brew --prefix openssl`/lib
+```
+
+Set up AWS credentials: environment variables (export AWS_ACCESS_KEY_ID and
+AWS_SECRET_ACCESS_KEY), populate the ~/.aws/credentials file, or use an
+IAM instance profile on an EC2 instance.
+
+Initialize and fetch the git submodule for botocore definitions:
+
+`git submodule init`
+
+`git submodule update`
+
+Initial setup complete, the above shouldn't be needed again unless you need to
+update the botocore definitions from that upstream project.
+
+## Building after initial setup
+
+Build the project with `cargo build`.
+
+`cargo test --verbose --features aws_integration` - This will create real AWS resources and you may be charged.
+
+For more verbose test output, you can run `cargo test --verbose --features aws_integration -- --nocapture`.
+
+## Rust code generation from boto core service definitions:
+
+See [CODEGEN](codegen/CODEGEN.md).

--- a/README.md
+++ b/README.md
@@ -53,15 +53,4 @@ Information on release schedules and procedures are in [RELEASING](RELEASING.md)
 
 ### Contributing
 
-1. Install Rust **1.3.0** or later - http://www.rust-lang.org/
-2. Check out code from github
-3. Set up AWS credentials: environment variables (export AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY), ~/.aws/credentials, or use an IAM instance profile.
-4. `cargo build`
-5. `cargo test --verbose --features aws_integration` - This will create real AWS resources and you may be charged.
-6. (For more verbose test output, you can run `cargo test --verbose --features aws_integration -- --nocapture` )
-
-If openssl isn't installed you may see an error compiling.  For OSX, running `brew install openssl` then `brew link --force openssl` should fix it.
-
-#### Rust code generation from boto core service definitions:
-
-See [CODEGEN](codegen/CODEGEN.md).
+See [CONTRIBUTING](CONTRIBUTING.md).

